### PR TITLE
create scroll listener on mount instead of exported singleton

### PR DESCRIPTION
**Pull request by jhsu**
the existing module exports an instance of `element-resize-detector` which tries to bind event listeners on `document.body` which may not be available at instantiation.

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [ ] IE11 (Win 7)
  - [ ] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- ~~[ ] One core team UX approval~~

### CODE REVIEW

**Comment by jhsu** [comment]
this creates a new instance which probably creates a new listener per instance of the resizer.

one possibility that I know react-waypoint uses is creates an event listener on the parent's dom node so that any sibling checks the parent for a listener before creating its own.

**Comment by sodiumjoe** [comment]
hm, generally there are at most a few resizing elements, right? Seems ok to me

File: src/components/Resizer/Resizer.spec.jsx [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+			removeListener: jest.fn(),
+		};
+	};
+});
 import _ from 'lodash';
 import React from 'react';
 import { common } from '../../util/generic-tests';
 import { mount } from 'enzyme';
 import Resizer from './Resizer';
-import { elementResizeDetector } from './Resizer.util';
```
Comment (sodiumjoe) [inline src/components/Resizer/Resizer.spec.jsx]: wasn't there some reason we had to do this in the first place @mute ?

Comment (mute) [inline src/components/Resizer/Resizer.spec.jsx]: We only pulled this out to a `util` to maintain the singleton while making testing a little easier.

If we're okay with it not being a singleton, we don't need it anymore.

**Pull request merged**